src/dune_rules/melange/melange_rules.ml

```diff
@@ -241,6 +241,64 @@ let setup_emit_cmj_rules ~sctx ~dir ~scope ~expander ~dir_contents
   in
   Buildable_rules.with_lib_deps ctx compile_info ~dir ~f
 
+module Runtime_deps = struct
+  let to_action_builder ~expander dep_conf =
+    let runtime_deps =
+      let runtime_deps, _sandbox = Dep_conf_eval.unnamed ~expander dep_conf in
+      runtime_deps
+    in
+    Action_builder.memoize "melange runtime deps" runtime_deps
+
+  let eval ~expander (deps : Dep_conf.t list) =
+    let open Memo.O in
+    let builder = to_action_builder ~expander deps in
+    let+ paths, _ = Action_builder.run builder Lazy in
+    paths
+
+  let targets ~output deps =
+    List.map
+      ~f:(fun src ->
+        let dst =
+          match output with
+          | `Private_library_or_emit output_dir ->
+            let target =
+              src |> Path.as_in_build_dir_exn
+              |> Path.Build.drop_build_context_exn
+            in
+            Path.Build.append_source output_dir target
+          | `Public_library (lib_dir, output_dir) ->
+            lib_output_path ~output_dir ~lib_dir src
+        in
+        (src, dst))
+      deps
+end
+
+let setup_runtime_assets_rules sctx ~dir ~target_dir ~mode
+    ~(mel : Melange_stanzas.Emit.t) ~output =
+  let open Memo.O in
+  let* melange_files =
+    let* expander = Super_context.expander sctx ~dir in
+    Runtime_deps.eval ~expander mel.runtime_deps
```

Comment:
I understand this kind of expansion doesn't support globs right?

Reply:
I'm pretty sure globs are supported. What made you think that?

Reply:
check out [01429cf](https://github.com/ocaml/dune/pull/7199/commits/01429cfb277cd04b70b40b861717b67b37fcbe5e) for an example of globs being supported.

Reply:
> I'm pretty sure globs are supported. What made you think that?

In [my attempt to implement runtime_deps](https://github.com/ocaml/dune/pull/6643), I took `Simple_rules.copy_files` as reference on how to handle globs. As one can see, that function handles a lot of edge cases like "Cannot find directory:" or "Cannot copy files onto themselves...". As I didn't see any of this handling in the PR, I assumed globs were not supported. Isn't it necessary to handle these cases here?

Reply:
oh interesting. I might need to do all of that indeed.